pkg/specgenutil/specgen.go

```diff
@@ -1297,3 +1301,293 @@ func GetResources(s *specgen.SpecGenerator, c *entities.ContainerCreateOptions)
 	}
 	return s.ResourceLimits, nil
 }
+
+func parseUint(val string) (uint, error) {
+	valUint, err := strconv.ParseUint(val, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return uint(valUint), nil
+}
+
+func GetChangedHealthCheckConfiguration(cmd *cobra.Command) define.UpdateHealthCheckConfig {
+	updateHealthCheckConfig := define.UpdateHealthCheckConfig{}
+	cmd.Flags().Visit(func(f *pflag.Flag) {
```

Comment:
I don't get why we loop here at all over all flags? Can't we just access all the flags directly if we have to list out all name anyway.

Reply:
Visit visits only those flags that have been set.

Reply:
you can use .Changed() on a flag to know if was set or not